src/main/java/org/jabref/logic/util/io/FileUtil.java

```diff
@@ -588,4 +588,73 @@ public static String shortenFileName(String fileName, Integer maxLength) {
     public static boolean isCharLegal(char c) {
         return Arrays.binarySearch(ILLEGAL_CHARS, c) < 0;
     }
+
+    public static boolean renameLinkedFileToName(LinkedFile linkedFile,
```

Comment:
This is too much logic to go untested, please unit-test this.

Reply:
You can call the already existing RenamePdfCleanup which does exactly this